packages/buidler-core/src/internal/buidler-evm/provider/node.ts

```diff
@@ -336,11 +337,16 @@ export class BuidlerNode {
 
     await this._addTransactionToBlock(block, tx);
 
-    const result = await this._vm.runBlock({
-      block,
-      generate: true,
-      skipBlockValidation: true
-    });
+    let result: RunBlockResult;
+    try {
+      result = await this._vm.runBlock({
+        block,
+        generate: true,
+        skipBlockValidation: true
+      });
+    } catch (error) {
+      throw new TransactionExecutionError(error.message);
+    }
```

Comment:
Also, shouldn't `runCall` have something similar?

Reply:
It should, it will improve the amount of information calling code has in case of failure.